src/ops_complex.rs

```diff
@@ -605,15 +600,25 @@ pub fn generate_permutation_matrix(
         32,
         stream,
     )?;
+    let (unique_variable_indexes, temp_storage) = temp_storage.split_at_mut(num_bytes);
+    let unique_variable_indexes = unsafe { unique_variable_indexes.transmute_mut() };
+    let (run_lengths, temp_storage) = temp_storage.split_at_mut(num_bytes);
+    let run_lengths = unsafe { run_lengths.transmute_mut() };
+    let (num_runs_out, encode_temp_storage) = temp_storage.split_at_mut(cell_size);
+    let num_runs_out = unsafe { &mut num_runs_out.transmute_mut()[0] };
     set_to_zero(run_lengths, stream)?;
     encode(
-        temp_storage,
+        encode_temp_storage,
         sorted_variable_indexes,
         unique_variable_indexes,
         run_lengths,
         num_runs_out,
         stream,
     )?;
+    let (run_offsets, run_indexes) = temp_storage.split_at_mut(num_cells * 4);
```

Comment:
nit `num_bytes` instead of `num_cells * 4`?

Reply:
with the latest changes, this line is no more